app/controllers/admin/categories_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Admin::CategoriesController < ApplicationController
+  before_action :admin_user
+
+  def index
+    @categories = Category.all
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      flash[:success] = "New category created!"
+      redirect_to admin_categories_path
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+    @category = Category.find(params[:id])
+
+  end
+
+  def update
+    @category = Category.find(params[:id])
+    if @category.update_attributes(category_params)
+      flash[:success] = "Category updated!"
+      redirect_to admin_categories_path
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @category = Category.find(params[:id])
+    if @category.destroy
+      flash[:success] = "Category deleted!"
+      redirect_to admin_categories_path
+    else
+      redirect_to admin_categories_path
+    end
```

Comment:
@shokogyu You can remove the `else` here, and then outside the `if condition` you can put `redirect_to admin_categories_path`